### DISCUSSION TITLE
problem with yii2

### DISCUSSION THREAD

**Discussion by Playys228** [westacks/telebot#84] [Q&A] [answered]
since i use yii2 when i try to start bot i have an issue with redeclaring env() function/

PHP Fatal error:  Cannot redeclare env() (previously declared in /app/vendor/illuminate/support/helpers.php:136) in 
/app/common/helpers.php on line 48
_________________________________________________________
/app/vendor/illuminate/support/helpers.php:136
if (! function_exists('env')) {
    /**
     * Gets the value of an environment variable.
     *
     * @param  string  $key
     * @param  mixed  $default
     * @return mixed
     */
    function env($key, $default = null)
    {
        return Env::get($key, $default);
    }
}
___________________________________________________________

___________________________________________________________
/app/common/helpers.php on line 48
function env($key, $default = null)
{

    $value = getenv($key) ?? $_ENV[$key] ?? $_SERVER[$key];

    if ($value === false) {
        return $default;
    }

    switch (strtolower($value)) {
        case 'true':
        case '(true)':
            return true;

        case 'false':
        case '(false)':
            return false;
    }

    return $value;
}
___________________________________________________________

If i comment the function here /app/vendor/illuminate/support/helpers.php:136
It is working fine but i think that's not a good approach maybe do you know something better what can I do ?

**Comment by punyflash** [comment]
Sadly, Laravel integrated into the bot core pretty hard. TBH, I was planing to make separate packages for Laravel and core, so there were no dependency conflicts, but there was no personal need for me to do so. You may try to somehow define env function before loading vendor packages. This is hacky solution, but only that can work

**Reply by Playys228** [reply]
Maybe do you know the way I can reach that result ? I haven't any ideas

**Reply by punyflash** [reply]
Sadly, no. But as you mentioned it, I think there is a need to release a v3 package that will not rely on Laravel at all. And for Laravel create separate package. Will consider doing that next month

**Reply by Playys228** [reply]
It will be cool, will be following to this. Thanks.

**Comment by punyflash** [comment, marked as answer]
There should be no problems with using Yii framework right now as Laravel dependencies was migrated to [`westacks/telebot-laravel`](https://github.com/westacks/telebot-laravel). Just use version `^3.x` of the package
